File: reports/Bondarenko/4/src/Task1/Notepad.java
import java.util.*;
import java.lang.StringBuilder;

public class Notepad {
    Map<Date, Note> entries;

    public Notepad() {
        entries = new HashMap<>();
    }

    public void addEntry(Date date, String text) {
        if (!entries.containsKey(date)) {
            entries.put(date, new Note());
        }

        Note entry = entries.get(date);
        entry.addNote(text);
    }

    public ArrayList<String> getEntryByDate(Date date) {
        if (!entries.containsKey(date)) {
            return new ArrayList<String>();
        }
        return entries.get(date).getNote();
    }

    @Override
    public String toString() {
        StringBuilder builder = new StringBuilder();
        entries.forEach((date, entry) -> {
            builder.append(date);
            builder.append("\n");
            ArrayList<String> notes = entry.getNote();
            for (String note : notes) {
                builder.append(note);
                builder.append("\n");
            }
        });

        return builder.toString();
    }

    private class Note {
        private ArrayList<String> entry;

        public Note() {
            entry = new ArrayList<String>();
        }

        public void addNote(String text) {
            entry.add(text);
        }

        public ArrayList<String> getNote() {
            return entry;
        }
    }
}

File: reports/Serko/4/src/Task1/Notepad.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace Task1
{
    public class Notepad
    {

        private class Note
        {
            public DateTime Date { get; }
            public string Text { get; }

            public Note(DateTime date, string text)
            {
                Date = date;
                Text = text;
            }
        }

        private Dictionary<DateTime, List<Note>> notes;

        public Notepad()
        {
            notes = new Dictionary<DateTime, List<Note>>();
        }

        public void AddNote(DateTime date, string text)
        {
            if (!notes.ContainsKey(date))
            {
                notes[date] = new List<Note>();
            }

            notes[date].Add(new Note(date, text));
        }

        public List<string> FindNotes(DateTime date)
        {
            if (notes.ContainsKey(date))
            {
                List<string> foundNotes = new List<string>();
                foreach (var note in notes[date])
                {
                    foundNotes.Add(note.Text);
                }
                return foundNotes;
            }
            else
            {
                return new List<string>();
            }
        }

        public void PrintAllNotes()
        {
            foreach (var entry in notes)
            {
                Console.WriteLine($"Date: {entry.Key.ToShortDateString()}");
                foreach (var note in entry.Value)
                {
                    Console.WriteLine($"- {note.Text}");
                }
                Console.WriteLine();
            }
        
    }

}
}
